src/csharp/Microsoft.Spark/Broadcast.cs

```diff
@@ -0,0 +1,159 @@
+        }
+
+        JvmObjectReference IJvmObjectReferenceProvider.Reference => _jvmObject;
+
+        /// <summary>
+        /// Get the broadcasted value.
+        /// </summary>
+        /// <returns>The broadcasted value</returns>
+        public object Value()
+        {
+            return BroadcastRegistry.s_registry[_bid];
+        }
+
+        /// <summary>
+        /// Asynchronously delete cached copies of this broadcast on the executors.
+        /// If the broadcast is used after this is called, it will need to be re-sent to each
+        /// executor.
+        /// </summary>
+        public void Unpersist()
+        {
+            _jvmObject.Invoke("unpersist");
+        }
+
+        /// <summary>
+        /// Delete cached copies of this broadcast on the executors. If the broadcast is used after
+        /// this is called, it will need to be re-sent to each executor.
+        /// </summary>
+        /// <param name="blocking">Whether to block until unpersisting has completed</param>
+        public void Unpersist(bool blocking)
+        {
+            _jvmObject.Invoke("unpersist", blocking);
+        }
+
+        /// <summary>
+        /// Destroy all data and metadata related to this broadcast variable. Use this with
+        /// caution; once a broadcast variable has been destroyed, it cannot be used again.
+        /// This method blocks until destroy has completed.
+        /// </summary>
+        public void Destroy()
+        {
+            _jvmObject.Invoke("destroy");
+        }
+
+        /// <summary>
+        /// Function that creates a file to store the broadcast value in the given path.
+        /// </summary>
+        /// <param name="sc">Spark Context object</param>
+        /// <param name="path">Path where file is to be created</param>
+        /// <param name="value">Broadcast value to be written to the file</param>
+        private void WriteBroadcastValueToFile(SparkContext sc, string path, object value)
+        {
+            Directory.CreateDirectory(sc.TempDir);
+            using FileStream f = File.Create(path);
+            Dump(value, f);
+        }
+
+        /// <summary>
+        /// Function that serializes and stores the object passed to the given Stream.
+        /// </summary>
+        /// <param name="value">Serializable object</param>
+        /// <param name="stream">Stream to which the object is serialized</param>
+        private void Dump(object value, Stream stream)
+        {
+            var formatter = new BinaryFormatter();
+            formatter.Serialize(stream, value);
+        }
+    }
+
+    /// <summary>
+    /// Registry to store the broadcast variables for access through workers.
+    /// </summary>
+    internal class BroadcastRegistry
+    {
+        public static ConcurrentDictionary<long, object> s_registry { get; set; }
+        public static ArrayList s_listBroadcastVariables { get; set; }
+
+        public BroadcastRegistry(IJvmBridge jvm)
```

Comment:
Every time this non-static constructor is called, it will reset the static properties.  Perhaps this should be a static constructor?  Or, you can just instantiate the properties like:

```
public static ArrayList s_listBroadcastVariables { get; } = new ConcurrentDictionary<long, object>();
```

Reply:
How can I access the `jvm` object to be able to do this?